sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -114,8 +119,21 @@ protected OperationInternalBase(ClientDiagnostics clientDiagnostics, Response ra
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <returns>The last HTTP response received from the server, including the final result of the long-running operation.</returns>
         /// <exception cref="RequestFailedException">Thrown if there's been any issues during the connection, or if the operation has completed with failures.</exception>
-        public virtual async ValueTask<Response> WaitForCompletionResponseAsync(CancellationToken cancellationToken) =>
-            await WaitForCompletionResponseAsync(DefaultPollingInterval, cancellationToken).ConfigureAwait(false);
+        public virtual async ValueTask<Response> WaitForCompletionResponseAsync(CancellationToken cancellationToken)
+        {
+            while (true)
```

Comment:
Is there a way to consolidate this loop.  All of the async ones use the same implementation and all of the sync ones use the same.

Reply:
I moved the logic into a separate `PollResponseAsync(pollingStrategy, cancellationToken)`.